webrender/src/util.rs

```diff
@@ -160,21 +162,114 @@ pub fn calculate_screen_bounding_rect(
     transform: &LayoutToWorldFastTransform,
     rect: &LayoutRect,
     device_pixel_scale: DevicePixelScale,
-) -> DeviceIntRect {
-    let points = [
-        transform.transform_point2d(&rect.origin),
-        transform.transform_point2d(&rect.top_right()),
-        transform.transform_point2d(&rect.bottom_left()),
-        transform.transform_point2d(&rect.bottom_right()),
+    screen_bounds: Option<&DeviceIntRect>,
+) -> Option<DeviceIntRect> {
```

Comment:
I recall this function as being extremely performance sensitive, so is there any way to have a fast path when it isn't necessary to do all of these calculations?

Reply:
The way it's build is to avoid extra steps unless necessary. So the old `transform_point2d` for each corner is equivalent to `transform_point_2d_homogeneous` followed by `to_point2d` at the end.
Extra overhead may come from the following sources:
  1. more stuff being moved around (copies are the doom of Rust code)
  2. comparing 4 f32 values to 0 in `homogens.iter().any(|h| h.w <= 0.0)`
  3. worse instruction cache utilization for the function body being grown

Overall, I think it's a reasonable compromise, and it's not clear to me if we can do better here.